core/__tests__/modules/codeConfig/codeConfig.ts

```diff
@@ -901,6 +901,119 @@ describe("modules/codeConfig", () => {
     });
   });
 
+  describe("models are properly locked in cli:config mode", () => {
+    beforeAll(async () => {
+      await helper.truncate();
+
+      process.env.GROUPAROO_RUN_MODE = "cli:config";
+      api.codeConfig.allowLockedModelChanges = true;
+      const { errors, seenIds, deletedIds } = await loadConfigDirectory(
+        path.join(__dirname, "..", "..", "fixtures", "codeConfig", "initial")
+      );
+      expect(errors).toEqual([]);
+      expect(seenIds).toEqual({
+        apikey: ["website_key"],
+        app: expect.arrayContaining(["data_warehouse", "events"]),
+        destination: ["test_destination"],
+        group: ["email_group", "high_value"],
+        property: expect.arrayContaining([
+          "user_id",
+          "email",
+          "last_name",
+          "first_name",
+        ]),
+        schedule: ["users_table_schedule"],
+        source: ["users_table"],
+        team: ["admin_team"],
+        teammember: ["demo"],
+      });
+      expect(deletedIds).toEqual({
+        apikey: [],
+        app: [],
+        destination: [],
+        group: [],
+        property: [],
+        schedule: [],
+        source: [],
+        team: [],
+        teammember: [],
+      });
+    });
+
+    afterAll(async () => {
+      await helper.truncate();
+      process.env.GROUPAROO_RUN_MODE = undefined;
+    });
+
+    test('settings are locked with "config:code"', async () => {
+      const setting = await plugin.readSetting("core", "cluster-name");
+      expect(setting.value).toBe("Test Cluster");
+      expect(setting.locked).toBe("config:code");
```

Comment:
There are still some models (settings, teams, apiKeys) that are always locked independent of running in config mode. Is the right behavior?

Reply:
Those are models that shouldn't be used in config mode... so that's an interesting question.  I think for now keeping them locked is OK.  If someone tries to write to them they will get an error which hints that we don't expect those models to be used right now